File: movingAvg.py
class MovingAverage(object):
    def __init__(self, size):
        self.size=size
        self.queue=[]
    def next(self,val):
        if not self.queue or len(self.queue)<self.size:
            self.queue.append(val)
        elif len(self.queue)==self.size:
            self.queue.pop(0)
            self.queue.append(val)
            return (float(sum(self.queue))/len(self.queue))
        return ('Append more values')

obj=MovingAverage(5)
print(obj.next(4))
print(obj.queue)
print(obj.next(9))
print(obj.queue)
print(obj.next(2))
print(obj.queue)
print(obj.next(9))
print(obj.queue)
print(obj.next(4))
print(obj.queue)
print(obj.next(0))

File: movingAvg.js
class MovingAverage{
    constructor(size){
        this.size=size
        this.queue=[]
    }
    next( val){
        if(this.queue.length<this.size){
            this.queue.push(val)
            return 'append more values'

        }
        else if(this.queue.length>=this.size){
            this.queue.shift()
            this.queue.push(val)
            return (this.queue.reduce((acc, curV) => acc+curV, 0)/this.queue.length)
        
}}
}
obj=new MovingAverage(5)
obj.next(3)
console.log(obj.queue)
obj.next(34)
console.log(obj.queue)
obj.next(36)
console.log(obj.queue)
obj.next(33)
console.log(obj.queue)

console.log(obj.next(2))

console.log(obj.next(23))

console.log(obj.next(29))